test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -0,0 +1,343 @@
+                                                                      force=True))
+        cli_messages.check_msg(output, cli_messages.remove_multilevel_core)
+
+    with TestRun.step("Stop cache."):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode.with_any_trait(CacheModeTrait.InsertRead
+                                                                | CacheModeTrait.InsertWrite))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_one_core_remove(cache_mode):
+    """
+        title: Test if OpenCAS correctly handles removal of one of multiple core devices.
+        description: |
+          When one core device is removed from a cache instance all blocks previously occupied
+          by the data from that core device should be removed. That means that the number of free
+          cache blocks should increase by the number of removed blocks.
+          Test is without pass through mode.
+        pass_criteria:
+          - No system crash.
+          - The remaining core is able to use cache.
+          - Removing core frees cache blocks occupied by this core.
+    """
+    with TestRun.step("Prepare one device for cache and two for core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(512, Unit.MebiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(1, Unit.GibiByte)] * 2)
+        core_part1 = core_dev.partitions[0]
+        core_part2 = core_dev.partitions[1]
+        Udev.disable()
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 2:
+            TestRun.fail(f"Expected cores count: 2; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Fill cache with pages from the first core."):
+        blocks = int(cache.size.value / block_size.value)
+        if CacheMode.get_traits(cache_mode) == CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input(core1.system_path)
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks))
+        elif CacheMode.get_traits(cache_mode) != CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input("/dev/urandom")
+                  .output(core1.system_path)
+                  .block_size(block_size)
+                  .count(blocks))
+        dd.run()
```

Comment:
Please move this to the separate method and use in each test with proper params (I suggest making method with 2 params: core_device and count)